### DISCUSSION TITLE
Allow PCSX-Redux users to add, use, disable and remove scriptpoints

### DISCUSSION THREAD

**Discussion by ghost** [grumpycoders/pcsx-redux#1076] [Ideas]
Every scriptpoint is associated with a script and when triggered the script is interpreted and executed.
The script can be written in any language that has an interpreter for like Lua.

A breakpoint is a special case of a scriptpoint where the script is to just pause the game.
A logpoint is another special case of a scriptpoint where the script is to print the value of PC to the Logs window.

Scriptpoint is the most **generic** and **general** concept.

What do you think that scriptpoints can be used for more than just pausing the game and print PC to Logs window?

**Comment by wheremyfoodat** [comment]
This is pretty much how breakpoints work already. If you read the wiki on the Lua API you'll see that the Lua API allows you to add breakpoints + a callback to be called when the breakpoint is triggered. The default action is to pause the emulator, but you can make it print the PC, or whatever.

**Reply by ghost** [reply]
> This is pretty much how breakpoints work already. If you read the wiki on the Lua API you'll see that the Lua API allows you to add breakpoints + a callback to be called when the breakpoint is triggered. The default action is to pause the emulator, but you can make it print the PC, or whatever.

Understood! I will learn the Lua API and master it!

So my request is now to add a new TextBox to the Breakpoints window/tab where the user can define the callback to be invoked on trigger then but this request can receive a low priority because what I want to do can already be done with the Lua API right now.
